rust/crd/src/listener.rs

```diff
@@ -111,8 +113,22 @@ pub fn get_kafka_listener_config(
         });
         listener_security_protocol_map
             .insert(KafkaListenerName::ClientAuth, KafkaListenerProtocol::Ssl);
+    } else if kafka_security.has_kerberos_enabled() {
+        // 2) Kerberos and TLS authentication classes are mutually exclusive
```

Comment:
optional: I think it would be great if we can assert this, e.g. using
```
        ensure!(
            kafka_security.tls_client_authentication_class().is_none(),
            FooErrorSnafu
        );
```

Reply:
You are right! We could add a different ensure!() instead, but it's not important. So feel free to ignore :)